src/main/java/at/pcgamingfreaks/mkvaudiosubtitlechanger/config/validator/ConfigValidator.java

```diff
@@ -0,0 +1,84 @@
+package at.pcgamingfreaks.mkvaudiosubtitlechanger.config.validator;
+
+import at.pcgamingfreaks.mkvaudiosubtitlechanger.config.Config;
+import at.pcgamingfreaks.mkvaudiosubtitlechanger.config.ValidationResult;
+import at.pcgamingfreaks.mkvaudiosubtitlechanger.model.ConfigProperty;
+import at.pcgamingfreaks.yaml.YAML;
+import at.pcgamingfreaks.yaml.YamlKeyNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.lang3.StringUtils;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+@RequiredArgsConstructor
+public abstract class ConfigValidator<FieldType> {
+	protected final ConfigProperty property;
+	protected final boolean required;
+	protected final FieldType defaultValue;
+
+    public ValidationResult validate(YAML yaml, CommandLine cmd) {
+		System.out.printf("Checking %s... ", property.prop());
+		String resultString = null;
+
+		if (cmd.hasOption(property.prop())) {
```

Comment:
Could this be abstacted by a method returning a Function<?, Fieldtype>
Maybe interate through a Collection of lists until value is found. Or overwrite previous values if value available (sort from highest to lowest prio and continue when value found)